subscription/src/main/java/org/killbill/billing/subscription/api/user/DefaultSubscriptionBaseApiService.java

```diff
@@ -669,6 +670,51 @@ public int handleBasePlanEvent(final DefaultSubscriptionBase subscription, final
         }
     }
 
+    @Override
+    public int handleExpiredEvent(final DefaultSubscriptionBase subscription, final SubscriptionBaseEvent event, final SubscriptionCatalog catalog, final CallContext context) throws CatalogApiException {
+
+        final InternalCallContext internalCallContext = createCallContextFromBundleId(subscription.getBundleId(), context);
+        if (event.getType() == EventType.EXPIRED && subscription.getCategory() == ProductCategory.BASE) {
+
+            final List<SubscriptionBaseEvent> expireEvents = new LinkedList<SubscriptionBaseEvent>();
+            final List<DefaultSubscriptionBase> subscriptionsToBeExpired = computeAddOnsToExpire(expireEvents, subscription.getBundleId(), event.getEffectiveDate(), catalog, internalCallContext);
+            dao.cancelSubscriptionsOnBasePlanEvent(subscription, event, subscriptionsToBeExpired, expireEvents, catalog, internalCallContext); //TODO_1533, not renamed this method to cancelOrExpireSubscriptionsOnBasePlanEvent as there are many other methods in the DAO.
+            return subscriptionsToBeExpired.size();
+        } else if (subscription.getCategory() == ProductCategory.STANDALONE || subscription.getCategory() == ProductCategory.ADD_ON) {
+            final List<SubscriptionBaseEvent> expireEvents = new LinkedList<SubscriptionBaseEvent>();
+            final List<DefaultSubscriptionBase> subscriptionsToBeExpired = new LinkedList<DefaultSubscriptionBase>();
+            dao.cancelSubscriptionsOnBasePlanEvent(subscription, event, subscriptionsToBeExpired, expireEvents, catalog, internalCallContext);
+            return 1;
+        } else {
```

Comment:
I don't think this `else` makes sense: We have 3 types of categories, and the if / else if statement above looked at all 3.

Reply:
Fixed, I've removed the last else statement